src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -699,7 +699,7 @@ codeBlockFenced = try $ do
               Just (elementId, classes, attrs) -> (elementId, maybe classes (: classes) languageId, attrs)))
   blankline
   contents <- T.intercalate "\n" <$>
-                 manyTill (gobbleAtMostSpaces indentLevel >> anyLine)
+                 manyTill (anyLine)
```

Comment:
Why do you consider this a bug?  It is meant to ensure that whitespace that only serves to align the code with the start of the fence will not be counted as part of the code.
e.g. in
````
1.  ``` ruby
    x = y
    ````
````
The contents of the code block should be `"x=y"` not `"    x=y"`.

Reply:
I think this PR is necessarily more destructive to pandoc than is feasible to get merged. The issue is that Agda's literate mode doesn't attempt to do any whitespace reindenting, so 

~~~
```agda
record Foo where
  field
```

some prose

1. ```agda
    someField : Type
    ```
~~~

gets mangled in the output to be

```agda
record Foo where
  field
someField : Type
```

which breaks the indent rules.

Reply:
I see.  Well, let's close this PR, then.  I think the Lua filter approach I outline above could be helpful, but you may run across the indentation problem with it too.